ngraph/frontend/paddlepaddle/src/op/lstm.cpp

```diff
@@ -0,0 +1,212 @@
+                                    ngraph::op::util::convert_lstm_node_format(
+                                            input_weight,
+                                            ngraph::op::util::LSTMWeightsFormat::IFCO,
+                                            ngraph::op::util::LSTMWeightsFormat::FICO,
+                                            1);
+                            m_input_map[LSTMInput::LSTM_INPUT_R] =
+                                    ngraph::op::util::convert_lstm_node_format(
+                                            hidden_weight,
+                                            ngraph::op::util::LSTMWeightsFormat::IFCO,
+                                            ngraph::op::util::LSTMWeightsFormat::FICO,
+                                            1);
+                            m_input_map[LSTMInput::LSTM_INPUT_B] =
+                                    ngraph::op::util::convert_lstm_node_format(
+                                            bias,
+                                            ngraph::op::util::LSTMWeightsFormat::IFCO,
+                                            ngraph::op::util::LSTMWeightsFormat::FICO,
+                                            1);
+
+                            // Get dimensions needed for default inputs creation
+                            // Parsing init hidden state
+                            auto shape_of_x = std::make_shared<opset6::ShapeOf>(input_x);
+
+                            auto axes =
+                                    opset6::Constant::create(element::Type_t::i32, Shape{1}, {0});
+
+                            auto batch_size_node = std::make_shared<opset6::Gather>(
+                                    shape_of_x,
+                                    opset6::Constant::create(element::Type_t::i32, Shape{1}, {0}),
+                                    axes);
+
+                            auto seq_length_node = std::make_shared<opset6::Gather>(
+                                    shape_of_x,
+                                    opset6::Constant::create(element::Type_t::i32, Shape{1}, {1}),
+                                    axes);
+
+                            auto shape_of_r = std::make_shared<opset6::ShapeOf>(
+                                    m_input_map[LSTMInput::LSTM_INPUT_R]);
+                            auto num_directions_node = std::make_shared<opset6::Gather>(
+                                    shape_of_r,
+                                    opset6::Constant::create(element::Type_t::i32, Shape{1}, {0}),
+                                    axes);
+                            auto hidden_size_node = std::make_shared<opset6::Gather>(
+                                    shape_of_r,
+                                    opset6::Constant::create(element::Type_t::i32, Shape{1}, {2}),
+                                    axes);
+
+                            m_input_map[LSTMInput::LSTM_INPUT_SEQ_LENGTHS] =
```

Comment:
PDPD LSTM has an optional `sequence_length ` input, please add TODO here to support it later.

Reply:
The sequence_length  is used only for constructing LSTM graph. When PDPD saving this model, the sequence_length is fused inside the model file. We can parse it from the Shape.

Reply:
"We can parse it from the Shape." - this is only for the case where seq_length is the same for each batch.
They can be different, It means we can execute different count of lstm cells for different batch of lstm sequence primitive.

Reply:
I check the code, the sequence_length will preform like a mask inside during the inference phase. I will add TODO here to support this in the future since current implementation can work with our target models.